corehq/apps/app_manager/tests/test_suite_inline_search.py

```diff
@@ -550,7 +616,12 @@ def test_child_module_with_inline_search_entry(self):
           <entry>
             <post url="http://localhost:8000/a/test_domain/phone/claim-case/"
                 relevant="count(instance('casedb')/casedb/case[@case_id=instance('commcaresession')/session/data/case_id_case]) = 0">
-              <data key="case_id" ref="instance('commcaresession')/session/data/case_id_case"/>
+              <data exclude="count(instance('casedb')/casedb/case[@case_id=instance('commcaresession')/session/data/case_id_case]) != 0"
+                key="case_id" ref="instance('commcaresession')/session/data/case_id_case"/>
+              <data exclude="count(instance('casedb')/casedb/case[@case_id=instance('commcaresession')/session/data/parent_id]) != 0"
+                key="case_id" ref="instance('commcaresession')/session/data/parent_id"/>
+              <data exclude="count(instance('casedb')/casedb/case[@case_id=instance('commcaresession')/session/data/case_id]) != 0"
+                key="case_id" ref="instance('commcaresession')/session/data/case_id"/>
```

Comment:
How come there are 3 data elements here? Looking at the session element there should only be `case_id_case` and `case_id`. Where does `parent_id` come from?

Reply:
`get_datum_meta_module` correctly returns the chain of parent select `session_var`. In this test, if submodule isn't being used, the session datums would have the IDs `parent_id` and `case_id`. By configuring the child parent select module as the submodule, `parent_id` changes to `case_id` and `case_id`  changes to `case_id_case`.

Still looking into where/why this happens. I'm suspecting it might be part of the post process, so I might have to change figuring out the parent datum ID to occur in the post process step as you suggested in a previous [comment](https://github.com/dimagi/commcare-hq/pull/33617#discussion_r1357944124).

Reply:
Ah, `get_case_datums_basic_module` is what creates the final case datums  [05683ae](https://github.com/dimagi/commcare-hq/pull/33738/commits/05683aecda28234bd997f8682bfd1ebdd660e1c2)